Show AndroidRuntime crashes from invalid AndroidManifest.xml in logs

The APK can fail to launch if the package name for the main activity is incorrect. This log seems to be filtered out, making the crash non-obvious. Instead the flutter tool hangs on "Waiting for observatory..." 

To reproduce this, `flutter create`  a new project and apply the following diff:

```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.example.foobar">
    <!-- io.flutter.app.FlutterApplication is an android.app.Application that
         calls FlutterMain.startInitialization(this); in its onCreate method.
         In most cases you can leave this as-is, but you if you want to provide
         additional functionality it is fine to subclass or reimplement
         FlutterApplication and put your custom class here. -->
    <application
---           android:name="io.flutter.app.FlutterApplication"
+++        android:name="io.flutter.app.FlutterApplication2"
        android:label="foobar"
        android:icon="@mipmap/ic_launcher">
        <activity
```

The resulting crash from logcat:

```
05-11 12:41:51.849  9709  9709 D AndroidRuntime: Shutting down VM
05-11 12:41:51.850  9709  9709 E AndroidRuntime: FATAL EXCEPTION: main
05-11 12:41:51.850  9709  9709 E AndroidRuntime: Process: com.example.foobar, PID: 9709
05-11 12:41:51.850  9709  9709 E AndroidRuntime: java.lang.RuntimeException: Unable to instantiate application io.flutter.app.FlutterApplication2: java.lang.ClassNotFoundException: Didn't find class "io.flutter.app.FlutterApplication2" on path: DexPathList[[zip file "/data/app/com.example.foobar-jjAiPTSBxSHLbAR1O2qlvQ==/base.apk"],nativeLibraryDirectories=[/data/app/com.example.foobar-jjAiPTSBxSHLbAR1O2qlvQ==/lib/arm64, /data/app/com.example.foobar-jjAiPTSBxSHLbAR1O2qlvQ==/base.apk!/lib/arm64-v8a, /system/lib64, /product/lib64]]
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at android.app.LoadedApk.makeApplication(LoadedApk.java:1226)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6431)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at android.app.ActivityThread.access$1300(ActivityThread.java:219)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:107)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:214)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7356)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
05-11 12:41:51.850  9709  9709 E AndroidRuntime: Caused by: java.lang.ClassNotFoundException: Didn't find class "io.flutter.app.FlutterApplication2" on path: DexPathList[[zip file "/data/app/com.example.foobar-jjAiPTSBxSHLbAR1O2qlvQ==/base.apk"],nativeLibraryDirectories=[/data/app/com.example.foobar-jjAiPTSBxSHLbAR1O2qlvQ==/lib/arm64, /data/app/com.example.foobar-jjAiPTSBxSHLbAR1O2qlvQ==/base.apk!/lib/arm64-v8a, /system/lib64, /product/lib64]]
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:196)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at android.app.AppComponentFactory.instantiateApplication(AppComponentFactory.java:76)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at androidx.core.app.CoreComponentFactory.instantiateApplication(CoreComponentFactory.java:47)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at android.app.Instrumentation.newApplication(Instrumentation.java:1155)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        at android.app.LoadedApk.makeApplication(LoadedApk.java:1218)
05-11 12:41:51.850  9709  9709 E AndroidRuntime:        ... 9 more
05-11 12:41:51.853  1412  6080 W ActivityTaskManager:   Force finishing activity com.example.foobar/.MainActivity
05-11 12:41:51.856  1412  9759 I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
```

--- a/packages/flutter_tools/lib/src/android/android_device.dart
+++ b/packages/flutter_tools/lib/src/android/android_device.dart
@@ -986,6 +986,7 @@ class AdbLogReader extends DeviceLogReader {
     RegExp(r'^[VDIWEF]\/flutter[^:]*:\s+', caseSensitive: false),
     RegExp(r'^[IE]\/DartVM[^:]*:\s+'),
     RegExp(r'^[WEF]\/AndroidRuntime:\s+'),
+    RegExp(r'^[WEF]\/AndroidRuntime\([0-9]+\):\s+'),
     RegExp(r'^[WEF]\/ActivityManager:\s+.*(\bflutter\b|\bdomokit\b|\bsky\b)'),
     RegExp(r'^[WEF]\/System\.err:\s+'),
     RegExp(r'^[F]\/[\S^:]+:\s+'),
@@ -1019,6 +1020,7 @@ class AdbLogReader extends DeviceLogReader {
     }
     final Match timeMatch = AndroidDevice._timeRegExp.firstMatch(line);
     if (timeMatch == null || line.length == timeMatch.end) {
+      _acceptedLastLine = false;
       return;
     }
     // Chop off the time.

--- a/packages/flutter_tools/test/general.shard/android/adb_log_reader_test.dart
+++ b/packages/flutter_tools/test/general.shard/android/adb_log_reader_test.dart
@@ -13,6 +13,11 @@ import '../../src/context.dart';
 const int kLollipopVersionCode = 21;
 const String kLastLogcatTimestamp = '11-27 15:39:04.506';
 
+/// By default the android log reader accepts lines that match no patterns
+/// if the previous line was a match. Include an intentionally non-matching
+/// line as the first input to disable this behavior.
+const String kDummyLine = 'Contents are not important\n';
+
 void main() {
   testWithoutContext('AdbLogReader calls adb logcat with expected flags apiVersion 21', () async {
     final FakeProcessManager processManager = FakeProcessManager.list(<FakeCommand>[
@@ -127,6 +132,40 @@ void main() {
 
     logReader.dispose();
     await onDone.future;
+  });
+
+  testWithoutContext('AdbLogReader does not filter output from AndroidRuntime crashes', () async {
+    final FakeProcessManager processManager = FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(
+        command: const <String>[
+          'adb',
+          '-s',
+          '1234',
+          'logcat',
+          '-v',
+          'time',
+        ],
+        completer: Completer<void>.sync(),
+        // Example stack trace from an incorrectly named application:name in the AndroidManfiest.xml
+        stdout:
+          kDummyLine +
+          '05-11 12:54:46.665 E/AndroidRuntime(11787): FATAL EXCEPTION: main\n'
+          '05-11 12:54:46.665 E/AndroidRuntime(11787): Process: com.example.foobar, PID: 11787\n'
+          '05-11 12:54:46.665 java.lang.RuntimeException: Unable to instantiate application '
+          'io.flutter.app.FlutterApplication2: java.lang.ClassNotFoundException:\n',
+      )
+    ]);
+    final AdbLogReader logReader = await AdbLogReader.createLogReader(
+      createMockDevice(null),
+      processManager,
+    );
+    await expectLater(logReader.logLines, emitsInOrder(<String>[
+      'E/AndroidRuntime(11787): FATAL EXCEPTION: main',
+      'E/AndroidRuntime(11787): Process: com.example.foobar, PID: 11787',
+      'java.lang.RuntimeException: Unable to instantiate application io.flutter.app.FlutterApplication2: java.lang.ClassNotFoundException:',
+    ]));
+
+    logReader.dispose();
   });
 }
 

